### ISSUE TITLE
[apex] SOQL performed in a for-each loop doesn't trigger ApexCRUDViolationRule

### ISSUE THREAD

**Issue by jfeingold35** [pmd/pmd#3462]
**Affects PMD Version:** 6.37.0

**Rule:** [ApexCRUDViolationRule](https://pmd.github.io/pmd-6.37.0/pmd_rules_apex_security.html#apexcrudviolation)

**Description:**
SOQL queries that occur inside a for-loop are not reported (false negative).
Seems to have been introduced by #3367; changing line 220 of `ApexCRUDViolationRule.java` to `return super.visit(node, data);` fixed the issue.

**Code Sample demonstrating the issue:**

```
public with sharing class Beep {
    public void boop() {
        Account[] accs = [SELECT Id FROM Account]; // This triggers the rule, like it should.
        for (Account acc : accs) {
            Account[] otherAccs = [SELECT Id FROM Account]; // This does NOT trigger the rule, but it should.
        }
    }
}
```

**Expected outcome:**
A violation should be reported at line 5, but isn't. False negative.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
CLI

**Comment by jfeingold35** [comment]
I'll be providing a patch presently.
